c_src/erl_optics.c

```diff
@@ -18,6 +18,8 @@ static ERL_NIF_TERM atom_dist;
 static ERL_NIF_TERM atom_quantile;
 static ERL_NIF_TERM atom_histo;
 static struct optics_poller *poller;
+static int carbon_poller;
+static int erlang_poller;
```

Comment:
I think these should just be booleans. Note that in see you can get access to the boolean type via `stdbool.h` header: https://en.cppreference.com/w/c/types/boolean